### PR TITLE
Inlcude batchID in instance/result file name

### PR DESCRIPTION
**Pull request by fleupold**
This PR makes it so that our instance/result files include the batchID and are also split up by date. We now create instance filenames in the following format:

`instances/{day}/instance_{batchId}_{datetime}.json`

and results in the following format:

`results/{day}/instance_{batchId}_{datetime}/<json|visualization>`

### Test Plan

Tested locally that files are created and read accordingly. For synchronisation with S3 I have to slightly adjust the sync script to not be too strict about the matching.

### CODE REVIEW

File: driver/src/price_finding/optimization_price_finder.rs [hunk 1]
```diff
@@ -223,9 +222,27 @@ impl PriceFinding for OptimisationPriceFinder {
             orders: orders.iter().map(From::from).collect(),
             fee: self.fee.as_ref().map(From::from),
         };
-        let current_time = Utc::now().to_rfc3339();
-        let input_file = format!("instances/instance_{}.json", &current_time);
-        let result_folder = format!("results/instance_{}/", &current_time);
+
+        let now = Utc::now();
+        let batch_id = now.timestamp() / 300;
```
Comment (e00E) [inline driver/src/price_finding/optimization_price_finder.rs]: Should it be `now.timestamp() / 300 - 1`?

File: driver/src/price_finding/optimization_price_finder.rs [hunk 2]
```diff
@@ -223,9 +222,27 @@ impl PriceFinding for OptimisationPriceFinder {
...
+        let input_file = format!(
+            "{}/instance_{}_{}.json",
+            &input_folder,
+            &batch_id,
+            &now.to_rfc3339()
+        );
+        let result_folder = format!(
+            "results/{}/instance_{}_{}/",
+            &date,
+            &batch_id,
```
Comment (e00E) [inline driver/src/price_finding/optimization_price_finder.rs]: To have this ordered lexicographically correctly we should 0 pad the batch_id.

**Review by e00E — APPROVED** [review]

File: docker-compose.yml [hunk 3]
```diff
@@ -35,3 +35,4 @@ services:
     volumes:
       # Sync src folder to allow development without rebuilding on each change
       - ./driver:/app/driver
+      - ./dex-contracts:/app/dex-contracts
```
Comment (nlordell) [inline docker-compose.yml]: Is this change related to the PR? I'm not sure I understand why it is needed for the instance and result file name changes.

File: driver/src/price_finding/optimization_price_finder.rs (hunk 1 above)
Comment (nlordell) [reply driver/src/price_finding/optimization_price_finder.rs]: Makes sense to me as we are solving for the previous batch.

File: driver/src/price_finding/optimization_price_finder.rs (hunk 2 above)
Comment (nlordell) [reply driver/src/price_finding/optimization_price_finder.rs]: While that is in general true, I don't think that is needed here as we are currently at batch `5279419` and until batch `10000000` is about 44 years (if my math isn't wrong).

**Review by nlordell — APPROVED** [review]
LGTM

File: driver/src/price_finding/optimization_price_finder.rs (hunk 1 above)
Comment (fleupold) [reply driver/src/price_finding/optimization_price_finder.rs]: Great catch!

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia0.giphy.com%2Fmedia%2F3o6ZsSRkXifoEJfpMA%2Fgiphy-downsized-medium.gif&token=great_catch" data-gitmeme-token="great_catch"></a>

File: driver/src/price_finding/optimization_price_finder.rs (hunk 2 above)
Comment (fleupold) [reply driver/src/price_finding/optimization_price_finder.rs]: It's a valid point, but I wouldn't know how many 0s I should use for padding and it's quite annoying to look at.

File: docker-compose.yml (hunk 3 above)
Comment (fleupold) [reply docker-compose.yml]: It's not required, but more convenient as otherwise you are stuck with the contract definitions from the time you built the container (might be outdated). And rebuilding just to get the contracts in there is time consuming. Will make this a separate PR though.
